app/src/main/java/io/gnosis/safe/ui/safe/settings/safe/SafeSettingsFragment.kt

```diff
@@ -26,11 +43,64 @@ class SafeSettingsFragment : BaseViewBindingFragment<FragmentSettingsSafeBinding
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.button.setOnClickListener {
-            viewModel.removeSafe()
+        binding.remove.setOnClickListener { showRemoveDialog() }
+        viewModel.state.observe(viewLifecycleOwner, Observer {
+            when (val viewAction = it.viewAction) {
+                is BaseStateViewModel.ViewAction.Loading -> loadDetails(viewAction.isLoading, it.safe, it.safeInfo, it.ensName)
+                is BaseStateViewModel.ViewAction.ShowError -> showError(viewAction.error)
+            }
+        })
+    }
+
+    private fun loadDetails(isLoading: Boolean, safe: Safe?, safeInfo: SafeInfo?, ensNameValue: String?) {
```

Comment:
you should split this function into showing loading and displaying state. Also loadDetails indicates that details will be loaded, but this function is about displaying stuff.